### DISCUSSION TITLE
subdag is not able to correctly render {{ next_ds }} jinja template when start_date uses timezone info.

### DISCUSSION THREAD

**Discussion by RajkotiyaDivyesh** [apache/airflow#21366] [General]
```
# ==========================
# dag_utc logs:
#  outer_test >>  cur_date: 2020-04-29, next_date: 2020-04-30, tomorrow_date: 2020-04-30, execution_date: 2020-04-29T05:00:00+00:00
#  inner_test >>  cur_date: 2020-04-29, next_date: 2020-04-30, tomorrow_date: 2020-04-30, execution_date: 2020-04-29T05:00:00+00:00
#
# dag_ist logs:
#   outer_test >>  cur_date: 2020-04-29, next_date: 2020-04-30, tomorrow_date: 2020-04-30, execution_date: 2020-04-29T05:00:00+00:00
#   inner_test >>  cur_date: 2020-04-29, next_date: 2020-04-29, tomorrow_date: 2020-04-30, execution_date: 2020-04-29T05:00:00+00:00
# ==========================
from datetime import datetime
import pendulum,logging
from airflow import DAG
from airflow.operators.python_operator import PythonOperator
from airflow.operators.subdag_operator import SubDagOperator
def get_test_operator(task_id):
    def fun(cdate, ndate, tdate, edate, **context):
        logging.info('cur_date: {}, next_date: {}, tomorrow_date: {}, execution_date: {}'.format(cdate, ndate, tdate, edate))
    return PythonOperator(task_id=task_id,
        python_callable=fun, provide_context=True,
        op_kwargs = {'cdate': '{{ds}}', 'ndate': '{{next_ds}}', 'tdate': '{{tomorrow_ds}}', 'edate': '{{execution_date}}'})
def test_subdag(parent_dag_name, start_date, schedule_interval):
    with DAG('%s.%s' % (parent_dag_name, 'child_dag'), schedule_interval=schedule_interval, start_date=start_date) as dag:
        task = get_test_operator('inner_test')
        return dag
with DAG('dag_utc', schedule_interval='0 5 * * *', start_date=datetime(2020,4,29)) as dag:
    task = get_test_operator('outer_test')
    sub_task = SubDagOperator(subdag=test_subdag(dag.dag_id, dag.start_date, dag.schedule_interval),task_id = 'child_dag')
    globals()['dag_utc'] = dag
with DAG('dag_ist', schedule_interval='30 10 * * *', start_date=datetime(2020,4,29, tzinfo=pendulum.timezone('Asia/Calcutta'))) as dag:
    task = get_test_operator('outer_test')
    sub_task = SubDagOperator(subdag=test_subdag(dag.dag_id, dag.start_date, dag.schedule_interval),task_id = 'child_dag')
    globals()['dag_ist'] = dag
```
**Apache Airflow version**: **v1.10.9**
hello airflow users,when i am trying to run this dags (dag_utc,dag_ist) somehow  for both dag in outer_test task '{{next_ds}}'  is  giving next execution date  but for  dag_ist(with 'Asia/Calcutta' timezone) in subdag(child_dag) '{{next_ds}}'  is giving current execution date for inner_test task.(here dag_ist execution_date is ( 2020-04-29T05:00:00+00:00) . can anyone explain is that a bug or  i am missing something?

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by ccecora** [comment]
@RajkotiyaDivyesh did you ever solve this issue?  Experiencing the same on my end in airflow 1.10.10.

**Comment by eladkal** [comment]
related https://github.com/apache/airflow/issues/14572

**Comment by uranusjr** [comment]
The displayed time here is actually pretty weird. `Asia/Calcutta` is +05:30 (`pendulum` seems to implement this correctly), so I'd expect the output to be `2020-04-29T05:30:00+00:00` (the same time as supplied by the user, converted to UTC). I'm less sure the two issues are related (still could be).

**Comment by kaxil** [comment]
Is this still an issue with the latest version of Airflow, can you please try and reproduce it?

**Comment by evillag** [comment]
It is an issue on 2.0.2 (AWS MWAA)

**Comment by potiuk** [comment]
I think it should be checked on a newer version. There were a number of things around that fixed in newer versions. MWAA also supports 2.2 Airlfow now. I will convert it into discussion now, but if you find it is still not fixed in those newer Airlfow version - do let us know.
